shared/test/test_animations.rb

```diff
@@ -0,0 +1,359 @@
+    # Make sure we have a clean starting point
+    delete_all_animation_versions(source_image_filename)
+    delete_all_animation_versions(dest_image_filename)
+
+    # Try to copy nonexistent source to destination
+    copy(source_image_filename, dest_image_filename)
+    assert not_found?
+  end
+
+  def test_animation_versions
+    filename = randomize_filename('test.png')
+    delete_all_animation_versions(filename)
+
+    # Create an animation file
+    v1_file_data = 'stub-v1-body'
+    upload(filename, v1_file_data, 'image/png')
+    assert successful?, @files_api.last_response.inspect
+
+    # Overwrite it.
+    v2_file_data = 'stub-v2-body'
+    upload(filename, v2_file_data, 'image/png')
+    assert successful?
+
+    # Delete it.
+    delete(filename)
+    assert successful?
+
+    # List versions.
+    versions = list_versions(filename)
+    assert successful?
+    assert_equal 2, versions.count
+
+    # Get the first and second version.
+    assert_equal v1_file_data, get_version(filename, versions.last['versionId'])
+    assert_equal v2_file_data, get_version(filename, versions.first['versionId'])
+
+    # Check cache headers
+    assert_equal 'public, max-age=3600, s-maxage=1800', @files_api.last_response['Cache-Control']
+  end
+
+  def test_replace_animation_version
+    filename = randomize_filename('replaceme.png')
+    delete_all_animation_versions(filename)
+
+    # Create an animation file
+    v1_file_data = 'stub-v1-body'
+    upload(filename, v1_file_data, 'image/png')
+    assert successful?
+    original_version_id = JSON.parse(@files_api.last_response.body)['versionId']
+
+    # Overwrite it, specifying the same version
+    v2_file_data = 'stub-v2-body'
+    upload_version(filename, original_version_id, v2_file_data, 'image/png')
+    new_version_id = JSON.parse(@files_api.last_response.body)['versionId']
+    assert successful?
+
+    # Make sure only one version exists
+    versions = list_versions(filename)
+    assert successful?
+    assert_equal 1, versions.count
+    assert_equal new_version_id, versions[0]['versionId']
+
+    # Note that even though we replaced a version, the version ID changed.
+    refute_equal original_version_id, new_version_id
+
+    # Make sure that one version has the newest content
+    assert_equal v2_file_data, get_version(filename, new_version_id)
+
+    delete(filename)
+  end
+
+  private
+
+  # Initialize test sessions for ChannelsApi and FilesApi
+  def init_apis
+    channels_api = Rack::Test::Session.new(Rack::MockSession.new(ChannelsApi, 'studio.code.org'))
+
+    # Make sure the animations api has the same storage id cookie used by the channels api.
+    channels_api.get '/v3/channels'
+    cookies = channels_api.last_response.headers['Set-Cookie']
+    files_mock_session = Rack::MockSession.new(FilesApi, "studio.code.org")
+    files_mock_session.cookie_jar.merge(cookies)
+    files_api = Rack::Test::Session.new(files_mock_session)
+
+    [channels_api, files_api]
+  end
```

Comment:
A better init strategy is to use `build_rack_mock_session` as shown in this PR: https://github.com/code-dot-org/code-dot-org/pull/7410 . Since you are basically copying AssetsTest and SourcesTest, I think it's ok to continue following the the init_apis pattern here. However, can you please add a comment to all 3 tests discouraging copying of `init_apis` and encouraging use of `build_rack_mock_session`?